nodedb.go

```diff
@@ -671,6 +702,41 @@ func (ndb *nodeDB) decrVersionReaders(version int64) {
 	}
 }
 
+func (ndb *nodeDB) flushVersion(version int64, rootHash []byte) error {
+	// Create new recentDB and snapshotDB batch objects. We don't use the current
+	// batch objects of the nodeDB as we don't want to write state prematurely or
+	// have conflicting state.
+	//
+	// NOTE: We ignore any write that happen to recentBatch.
+	rb := ndb.recentDB.NewBatch()
+	sb := ndb.snapshotDB.NewBatch()
+
+	// save branch, the root, and the necessary orphans
+	node := ndb.GetNode(rootHash)
+	ndb.savebranchBatch(node, true, rb, sb)
+
+	traverseOrphansVersionFromDB(ndb.recentDB, version, func(k, v []byte) {
+		var fromVersion, toVersion int64
+		orphanKeyFormat.Scan(k, &toVersion, &fromVersion)
+
+		ndb.saveOrphan(v, fromVersion, toVersion, true)
+	})
+
+	sb.Set(ndb.rootKey(version), node.hash)
+
+	if ndb.opts.Sync {
```

Comment:
This is sort of surprising to me, I would expect a `flush` method to flush the data all the way to disk (without sync, the OS buffers the write in memory). There could perhaps be a parameter `forceSync` or something.

Reply:
Indeed, good catch. This logic was based on what is found in `nodeDB#Commit`. However, since we want to flush to disk, I see no reason to consider the non-`ndb.opts.Sync` case -- we should always use `WriteSync`. Correct me if I'm wrong.

Reply:
I think that makes sense.